### PR TITLE
Added login page

### PR DESCRIPTION
**Pull request by tonyco97**

### CODE REVIEW

File: pages/login.tsx [hunk 1]
```diff
@@ -1,22 +1,103 @@
 // Copyright (C) 2022 Nethesis S.r.l.
 // SPDX-License-Identifier: AGPL-3.0-or-later
 
-import type { NextPage } from 'next'
-import * as Janus from 'janus-gateway'
-import { MdFavoriteBorder } from 'react-icons/md'
+import Image from 'next/image'
+import { useRouter } from 'next/router'
+import { useState } from 'react'
+import { TextInput } from '../components/common'
```
Comment (SebastianMB-IT) [inline pages/login.tsx:7]: This import with the one below can be merged.

File: pages/login.tsx [hunk 2]
```diff
@@ -1,22 +1,103 @@
...
+import { Button } from '../components/common'
+import hmacSHA256 from 'crypto-js/hmac-sha256'
+import Base64 from 'crypto-js/enc-base64'
+import { MdLock } from 'react-icons/md'
```
Comment (SebastianMB-IT) [inline pages/login.tsx:11]: The same thing as the comment above.

File: pages/login.tsx [hunk 3]
```diff
@@ -1,22 +1,103 @@
...
-  console.log(Janus)
+export default function Login() {
+  const [pwdVisible, setPwdVisible] = useState(false)
+  const router = useRouter()
+  const server = 'https://nethctiapp.nethserver.net'
+
+  const doLogin = async (e: React.FormEvent) => {
+    e.preventDefault()
+    const target = e.target as HTMLFormElement
+    const username = (target.querySelector('input[name="username"]') as HTMLInputElement).value
```
Comment (SebastianMB-IT) [inline pages/login.tsx:22]: Could you use refs in React instead querySelectors?

**Review by SebastianMB-IT — CHANGES_REQUESTED** [review]
Resolve the comments in the code and remember to add alerts in case of wrong password or other errors.

File: pages/login.tsx [hunk 4]
```diff
@@ -1,22 +1,112 @@
...
+      }),
+    })
+
+    const callStatus = res.status
+    const nonce = res.headers.get('www-authenticate')?.split(' ')[0]
+    const token = nonce
+      ? Base64.stringify(hmacSHA256(`${username}:${password}:${nonce}`, password))
+      : ''
+    if (token) {
+      router.push('/')
```
Comment (SebastianMB-IT) [inline pages/login.tsx:43]: The token must be saved inside the browser!

**Review by SebastianMB-IT — CHANGES_REQUESTED** [review]

File: pages/login.tsx [hunk 5]
```diff
@@ -1,22 +1,134 @@
...
+  const [onError, setOnError] = useState(false)
+  const router = useRouter()
+  const usernameRef = useRef() as React.MutableRefObject<HTMLInputElement>
+  const passwordRef = useRef() as React.MutableRefObject<HTMLInputElement>
 
-  return (
-    <>
-      <h1 className='text-5xl font-extrabold text-sky-600 flex gap-6'>
-        <p>Login</p>
+  function checkToken() {
```
Comment (SebastianMB-IT) [inline pages/login.tsx:21]: Put this function inside the lib directory.

File: pages/login.tsx [hunk 6]
```diff
@@ -1,22 +1,134 @@
...
+    let savedToken = localStorage.getItem('credentials')
+    if (!savedToken) {
+      return 'empty'
+    }
+  }
+
+  function setToken(token: any) {
```
Comment (SebastianMB-IT) [inline pages/login.tsx:28]: Do the same as above.

**Review by SebastianMB-IT — CHANGES_REQUESTED** [review]

**Review by SebastianMB-IT — APPROVED** [review]

**Review by andre8244 — APPROVED** [review]

**Pull request merged**